src/widgets/container.cpp

```diff
@@ -162,4 +161,35 @@ namespace gcn
     {
         return Rectangle(0, 0, getWidth(), getHeight());
     }
+
+    void Container::drawChildren(Graphics* graphics)
+    {
+        graphics->pushClipArea(getChildrenArea());
+
+        std::list<Widget*> children = getChildren();
+        std::list<Widget*>::iterator iter;
+        for (iter = children.begin(); iter != children.end(); ++iter)
```

Comment:
for range is possible here

Reply:
Updated with a for range loop. Also fixed some compiler warnings about types mismatch there (int vs unsigned int)